cmd/autoheal/alert_worker_test.go

```diff
@@ -0,0 +1,149 @@
+/*
+Copyright (c) 2018 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"path/filepath"
+	"testing"
+
+	"github.com/openshift/autoheal/pkg/alertmanager"
+	"github.com/openshift/autoheal/pkg/apis/autoheal"
+	batch "k8s.io/api/batch/v1"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+)
+
+func TestPickAlert(t *testing.T) {
+	file := filepath.Join("..", "..", "testdata", "empty-config.yml")
+	healer, err := NewHealerBuilder().
+		ConfigFile(file).
+		Build()
+
+	if err != nil {
+		t.Error(err)
+	}
+
+	defer runtime.HandleCrash()
+
+	alert := &alertmanager.Alert{
+		Status: "firing",
+		Labels: map[string]string{
+			"mylabel": "myvalue",
+		},
+	}
+
+	healer.alertsQueue.Add(alert)
+
+	// one alert in queue.
+	if healer.pickAlert() != true {
+		t.Errorf("Expected pickAlert to return true (i.e. there is an alert in alertQueue) but got false")
+	}
+}
+
+func TestStartHealing(t *testing.T) {
+	file := filepath.Join("..", "..", "testdata", "empty-config.yml")
+	healer, err := NewHealerBuilder().
+		ConfigFile(file).
+		Build()
+
+	healer.actionRunners[ActionRunnerTypeAWX] = FakeAWXActionRunner{}
+	healer.actionRunners[ActionRunnerTypeBatch] = FakeBatchActionRunner{}
+
+	if err != nil {
+		t.Error(err)
+	}
+
+	// Test 0
+
+	alert0 := &alertmanager.Alert{
+		Status: "firing",
+		Labels: map[string]string{
+			"mylabel": "myvalue",
+		},
+	}
+
+	change0 := &RuleChange{
+		Type: watch.Added,
+		Rule: &autoheal.HealingRule{
+			ObjectMeta: meta.ObjectMeta{
+				Name: "test-rule",
+			},
+			Labels: map[string]string{
+				"mylabel": "myvalue",
+			},
+			AWXJob: &autoheal.AWXJobAction{
+				Template: "Test AWX JOB",
+			},
+		},
+	}
+
+	// Add the rule change to rulesCache
+	healer.processRuleChange(change0)
+
+	output := healer.startHealing(alert0)
+
+	expected := fmt.Sprintf("Fake AWXActionRunner called with rule '%s' and alert '%s'",
+		change0.Rule.ObjectMeta.Name,
+		alert0.Name(),
+	)
+
+	if output.Error() != expected {
+		t.Errorf("Expected to get error: %s, instead got %s", expected, output.Error())
+	}
+
+	// Test 1
```

Comment:
I think, but I'm not sure, that you can separate these tests to two functions. Otherwise the failure of the first test might hide another failure of the second test - but I'm not 100% sure on the standard practice for this in Go, so don't take my word for it

Reply:
Hmm we could use a test table (see [this](https://nathanleclaire.com/blog/2015/10/10/interfaces-and-composition-for-effective-unit-testing-in-golang/) article on testing in GO.) but it's a bit of an over kill for two tests.